src/identity-providers/add/AddMapper.tsx

```diff
@@ -54,11 +56,21 @@ export const AddMapper = () => {
   const { providerId, alias } = useParams<IdentityProviderAddMapperParams>();
   const { id } = useParams<IdentityProviderEditMapperParams>();
 
-  const isSAMLorOIDC = providerId === "saml" || providerId === "oidc";
+  const identityProviders = _.groupBy(
+    useServerInfo().identityProviders,
+    "groupName"
+  );
```

Comment:
Since this operation might possibly be intensive to do you could memoize it so it does not get recalculated each time the component is rendered.
```suggestion
  const serverInfo = useServerInfo();
  const identityProviders = useMemo(
    () => groupBy(serverInfo.identityProviders, "groupName"),
    [serverInfo]
  );
```